sdk/core/azure-core/inc/azure/core/internal/strings.hpp

```diff
@@ -33,8 +33,10 @@ namespace Azure { namespace Core { namespace _internal {
     static bool LocaleInvariantCaseInsensitiveEqual(
         const std::string& lhs,
         const std::string& rhs) noexcept;
-    static std::string const ToLower(const std::string& src) noexcept;
-    static unsigned char ToLower(const unsigned char src) noexcept;
+    static std::string const ToLower(std::string const& src) noexcept;
+    static unsigned char ToLower(unsigned char const src) noexcept;
+    static std::string const ToUpper(std::string const& src) noexcept;
+    static unsigned char ToUpper(unsigned char const src) noexcept;
```

Comment:
This API addition, as a complement to `ToLower` makes sense to add.

However, unlike private APIs in the `_detail` namespace, we cannot make breaking changes to things in `_internal`. Therefore, we typically only add public surface area to such headers within the SDK/Azure Core when an actual upstream service SDK package within the repo needs it. In this case, this looks to only be used by unit tests.

In general, we'd avoid adding APIs to the SDK that are only used by unit tests :)